CMakeLists.txt

```diff
@@ -96,51 +135,81 @@ add_subdirectory(source/MaterialXCore)
 add_subdirectory(source/MaterialXFormat)
 
 # Add shader generation subdirectories
-add_subdirectory(source/MaterialXGenShader)
-add_subdirectory(source/MaterialXGenOsl)
-add_subdirectory(source/MaterialXGenGlsl)
+if (MATERIALX_BUILD_GEN)
+    # TODO: split up libraries to only install those relevant to enabled generators
+    add_subdirectory(libraries)
+    add_subdirectory(source/MaterialXGenShader)
+    if (MATERIALX_BUILD_GEN_GLSL)
+        add_subdirectory(source/MaterialXGenGlsl)
+    endif()
+    if (MATERIALX_BUILD_GEN_OSL)
+        add_subdirectory(source/MaterialXGenOsl)
+    endif()
+endif()
 
 # Add render module subdirectories
-if(MATERIALX_BUILD_OIIO)
-    add_definitions(-DMATERIALX_BUILD_OIIO)
-    add_definitions(-DOPENIMAGEIO_ROOT_DIR=\"${OPENIMAGEIO_ROOT_DIR}\")
-endif()
-add_subdirectory(source/MaterialXRender)
-add_subdirectory(source/MaterialXRenderOsl)
-add_subdirectory(source/MaterialXRenderHw)
-add_subdirectory(source/MaterialXRenderGlsl)
-
-# Add viewer subdirectory
-if(MATERIALX_BUILD_VIEWER)
-    add_subdirectory(source/MaterialXView)
-endif()
+if (MATERIALX_BUILD_RENDER)
+    if(MATERIALX_BUILD_OIIO)
+        add_definitions(-DMATERIALX_BUILD_OIIO)
+        add_definitions(-DOPENIMAGEIO_ROOT_DIR=\"${OPENIMAGEIO_ROOT_DIR}\")
+    endif()
+    add_subdirectory(source/MaterialXRender)
+    add_subdirectory(source/MaterialXRenderOsl)
+    add_subdirectory(source/MaterialXRenderHw)
+    add_subdirectory(source/MaterialXRenderGlsl)
+    add_subdirectory(resources)
+
+    if(MATERIALX_TEST_RENDER)
+        add_definitions(-DMATERIALX_TEST_RENDER)
+    endif()
 
-# Add test subdirectory
-if(MATERIALX_TEST_RENDER)
-    add_definitions(-DMATERIALX_TEST_RENDER)
+    # Add viewer subdirectory
+    if(MATERIALX_BUILD_VIEWER)
+        add_subdirectory(source/MaterialXView)
+    endif()
 endif()
-add_subdirectory(source/MaterialXTest)
-
-# Add resource and library subdirectories
-add_subdirectory(libraries)
-add_subdirectory(resources)
 
 if(MATERIALX_BUILD_PYTHON)
     add_subdirectory(source/PyMaterialX)
     add_subdirectory(python)
 endif()
+
 if(MATERIALX_BUILD_DOCS)
     add_subdirectory(documents)
 endif()
 
+if(MATERIALX_BUILD_TESTS)
+    add_subdirectory(source/MaterialXTest)
+endif()
+
 if(${CMAKE_VERSION} VERSION_GREATER "3.6.2")
     if(MATERIALX_BUILD_VIEWER)
         set_property(DIRECTORY PROPERTY VS_STARTUP_PROJECT MaterialXView)
     else()
-        set_property(DIRECTORY PROPERTY VS_STARTUP_PROJECT MaterialXTest)
+        if (NOT MATERIALX_DISABLE_BUILD_RENDER)
```

Comment:
Should this check instead be for MATERIALX_BUILD_TESTS?  I'm imagining that we'd want MaterialXTest to be the startup project in any solution where it's present without MaterialXView, but I may be missing a potential use case.